### ISSUE TITLE
Add caching, flushing, and minification to nav and page head

### ISSUE THREAD

**Issue by calvinf** [calvinf/versesforlife#45]
As a user, I want first paint to happen as fast as possible.  Right now, I'm not outputting anything from Node.js & Express (calling render) until the whole page and all views are ready to output.

* [x] Minify `<head>` (this could be accomplished by using `preact-helmet` to generate the `<head>` content
* [x] Add some kind of in-memory cache for `<head>` and `<nav>` based on their inputs
  * store in Redis or something?
* [ ] Flush after `<head>`
* [ ] Flush after `<nav>` in `<body>`

**Comment by calvinf** [comment]
Consider [express-stream](https://www.npmjs.com/package/express-stream).

**Comment by calvinf** [comment]
78eaf85 sets it up such that we're streaming back the response.  We can take this a step further by by using `openHtmlOpenHead`, rendering the `title` as quick as possible, and flushing out the entire `<head>` and `<nav>` view before even touching rendering of the main route view.  Some of these components can be highly cached as well -- particularly the `<head>` where the only thing that changes between renders is the title.

**Comment by calvinf** [comment]
GZIP compression seems to be missing with initial attempts at using express-stream.  I'll need to try a bit more locally -- on the `stream` branch -- to see if I can figure out what it's doing and if I can use it.  Otherwise I may be able to take a different approach using normal express functionality such as sending multiple chunks down using `res.send()` and then calling `res.close()` at the end.

**Comment by calvinf** [comment]
We have per-route caching and we're doing heavy caching of verse and pack content.  Current version of express-stream doesn't support gzip (apparently) and there aren't any other great solutions that I've been able to find.  Given that the page length is pretty small and not generally blocked for long, I'm going to resolve this for now since we got Preact in place for minification, we got version hashes in place for CSS & JS URLs, and there are other more immediate performance improvements we can pursue like bulk pack caching, etc.
